src/Servers/Kestrel/shared/test/TransportTestHelpers/TestServer.cs

```diff
@@ -28,7 +28,7 @@ namespace Microsoft.AspNetCore.Server.Kestrel.FunctionalTests
     /// </summary>
     internal class TestServer : IDisposable, IStartup
```

Comment:
We should make TestServer implement IAsyncDisposable, so we can just change our `using`s to `await using`s. Even though IHost doesn't implement IAsyncDisposable, [Host does](https://github.com/dotnet/runtime/blob/f0ede2b86b0f1b86744aa1e020ba1df3ac3d2744/src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs#L15) so we can just cast.

Surprisingly, it doesn't look like disposing the Host actually stops it when it's running, but I haven't tested that. Not a big deal either way as we can always call Host.StopAsync() before calling Host.DispooseAsync() in TestServer.DisposeAsync().

Reply:
Should I add `Host.StopAsync()` in `TestServer.DisposeAsync()`? Right now I call `server.StopAsync()` for every instance of `TestServer` before it is disposing.

Reply:
That's what we did in SignalR tests. It would clean up all the tests a little :D

Reply:
Ok. Time to revert some commits :grin: I'll remove the `server.StopAsync` from the tests.

Reply:
Should I just remove `StopAsync()` method in `TestServer`?